partial aggregation pushdown fails in TestHivePushdownFilterQueries and TestHivePushdownDistributedQueries

If enable partial aggregation in TestHivePushdownFilterQueries and TestHivePushdownIntegrationSmokeTest:
```
DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(getTables(),
                ImmutableMap.of("experimental.pushdown-subfields-enabled", "true", "experimental.pushdown-dereference-enabled", "true"),
                "sql-standard",
                ImmutableMap.of("hive.pushdown-filter-enabled", "true", "hive.enable-parquet-dereference-pushdown", "true", "hive.partial_aggregation_pushdown_enabled", "true", "hive.partial_aggregation_pushdown_for_variable_length_datatypes_enabled", "true"),
                Optional.empty());

queryRunner.execute(noPushdownFilter(queryRunner.getDefaultSession()),
```

                    ImmutableList.of(ORDERS, CUSTOMER, LINE_ITEM, PART_SUPPLIER),
                    ImmutableMap.of("experimental.pushdown-subfields-enabled", "true", "experimental.pushdown-dereference-enabled", "true"),
                    "sql-standard",
                    ImmutableMap.of("hive.pushdown-filter-enabled", "true", "hive.enable-parquet-dereference-pushdown", "true"),
                    ImmutableMap.of("hive.pushdown-filter-enabled", "true", "hive.enable-parquet-dereference-pushdown", "true", "hive.partial_aggregation_pushdown_enabled", "true", "hive.partial_aggregation_pushdown_for_variable_length_datatypes_enabled", "true"),
                    Optional.empty()),
                createBucketedSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))),
                createMaterializeExchangesSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))),
```

will get following error:
[ERROR] testAggregateSingleColumn(com.facebook.presto.hive.TestHivePushdownIntegrationSmokeTest)  Time elapsed: 2.194 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT MAX(comment) FROM orders
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:134)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:129)
	at com.facebook.presto.tests.AbstractTestIntegrationSmokeTest.testAggregateSingleColumn(AbstractTestIntegrationSmokeTest.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 17 more
Caused by: java.lang.UnsupportedOperationException: No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.writeMinMax(AggregatedOrcPageSource.java:172)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:116)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_061017_4.run(Unknown Source)
	... 3 more

[ERROR] testNullBucket(com.facebook.presto.hive.TestHivePushdownIntegrationSmokeTest)  Time elapsed: 0.25 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT COUNT() FROM table_with_null_buckets WHERE key IS NULL
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:144)
	at com.facebook.presto.hive.TestHiveIntegrationSmokeTest.testNullBucket(TestHiveIntegrationSmokeTest.java:2185)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 16 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_061017_4.run(Unknown Source)
	... 3 more

[ERROR] testBooleans(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 5.629 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE is_returned is not null
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testBooleans(TestHivePushdownFilterQueries.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 18 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062606_5113.run(Unknown Source)
	... 3 more

[ERROR] testBytes(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 8.1 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE ship_day > 15
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testBytes(TestHivePushdownFilterQueries.java:221)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 18 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062552_5111.run(Unknown Source)
	... 3 more

[ERROR] testDecimals(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 53.143 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE discount_long_decimal IS NOT NULL
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertFilterProject(TestHivePushdownFilterQueries.java:701)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testDecimals(TestHivePushdownFilterQueries.java:390)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 19 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062552_5111.run(Unknown Source)
	... 3 more

[ERROR] testDouble(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 6.153 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE discount IS NULL
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertFilterProject(TestHivePushdownFilterQueries.java:701)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testDouble(TestHivePushdownFilterQueries.java:292)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 19 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062606_5113.run(Unknown Source)
	... 3 more

[ERROR] testFloats(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 37.422 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE tax_real IS NOT NULL
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertFilterProject(TestHivePushdownFilterQueries.java:701)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testFloats(TestHivePushdownFilterQueries.java:320)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 19 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062552_5111.run(Unknown Source)
	... 3 more

[ERROR] testNumeric(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 0.726 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM orders WHERE orderkey BETWEEN 100 AND 1000 AND custkey BETWEEN 500 AND 800
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:134)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:129)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testNumeric(TestHivePushdownFilterQueries.java:246)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 17 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062611_5114.run(Unknown Source)
	... 3 more

[ERROR] testRcAndTextFormats(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 39.12 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT COUNT(*) FROM test_file_format_orc
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1129)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1124)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertFileFormat(TestHivePushdownFilterQueries.java:1037)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testRcAndTextFormats(TestHivePushdownFilterQueries.java:970)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: Error opening Hive split file:/tmp/PrestoTest7205970413479009931/hive_data/tpch/test_file_format_orc/ds=2019-11-01/20200910_063440_00217_pwbxv_944cfbcb-d6cf-4130-ba2a-8a486a969456 (offset=0, length=645326): -20
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 19 more
Caused by: com.facebook.presto.spi.PrestoException: Error opening Hive split file:/tmp/PrestoTest7205970413479009931/hive_data/tpch/test_file_format_orc/ds=2019-11-01/20200910_063440_00217_pwbxv_944cfbcb-d6cf-4130-ba2a-8a486a969456 (offset=0, length=645326): -20
	at com.facebook.presto.hive.rcfile.RcFilePageSourceFactory.createPageSource(RcFilePageSourceFactory.java:177)
	at com.facebook.presto.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:376)
	at com.facebook.presto.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:170)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:63)
	at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:80)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:248)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062558_5112.run(Unknown Source)
	... 3 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: -20
	at com.facebook.presto.rcfile.RcFileReader.<init>(RcFileReader.java:216)
	at com.facebook.presto.rcfile.RcFileReader.<init>(RcFileReader.java:118)
	at com.facebook.presto.hive.rcfile.RcFilePageSourceFactory.createPageSource(RcFilePageSourceFactory.java:150)
	... 16 more

[ERROR] testStrings(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 14.481 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE comment is NULL
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertFilterProject(TestHivePushdownFilterQueries.java:701)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testStrings(TestHivePushdownFilterQueries.java:419)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 19 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062552_5111.run(Unknown Source)
	... 3 more

[ERROR] testTimestamps(com.facebook.presto.hive.TestHivePushdownFilterQueries)  Time elapsed: 3.316 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE ship_timestamp IS NOT NULL
	at org.testng.Assert.fail(Assert.java:83)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1139)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.assertQueryUsingH2Cte(TestHivePushdownFilterQueries.java:1134)
	at com.facebook.presto.hive.TestHivePushdownFilterQueries.testTimestamps(TestHivePushdownFilterQueries.java:267)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: No value present
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:446)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 18 more
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.hive.orc.AggregatedOrcPageSource.getNextPage(AggregatedOrcPageSource.java:100)
	at com.facebook.presto.operator.TableScanOperator.getOutput(TableScanOperator.java:251)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:382)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:284)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:672)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:277)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
	at com.facebook.presto.$gen.Presto_0_242_SNAPSHOT_8b1f15d__testversion____20200910_062558_5112.run(Unknown Source)
	... 3 more

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestHivePushdownFilterQueries.testBooleans:197->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE is_returned is not null
[ERROR]   TestHivePushdownFilterQueries.testBytes:221->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE ship_day > 15
[ERROR]   TestHivePushdownFilterQueries.testDecimals:390->assertFilterProject:701->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE discount_long_decimal IS NOT NULL
[ERROR]   TestHivePushdownFilterQueries.testDouble:292->assertFilterProject:701->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE discount IS NULL
[ERROR]   TestHivePushdownFilterQueries.testFloats:320->assertFilterProject:701->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE tax_real IS NOT NULL
[ERROR]   TestHivePushdownFilterQueries.testNumeric:246->AbstractTestQueryFramework.assertQuery:129->AbstractTestQueryFramework.assertQuery:134 Execution of 'actual' query failed: SELECT count(*) FROM orders WHERE orderkey BETWEEN 100 AND 1000 AND custkey BETWEEN 500 AND 800
[ERROR]   TestHivePushdownFilterQueries.testRcAndTextFormats:970->assertFileFormat:1037->assertQueryUsingH2Cte:1124->assertQueryUsingH2Cte:1129->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT COUNT(*) FROM test_file_format_orc
[ERROR]   TestHivePushdownFilterQueries.testStrings:419->assertFilterProject:701->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE comment is NULL
[ERROR]   TestHivePushdownFilterQueries.testTimestamps:267->assertQueryUsingH2Cte:1134->assertQueryUsingH2Cte:1139->AbstractTestQueryFramework.assertQuery:139 Execution of 'actual' query failed: SELECT count(*) FROM lineitem_ex WHERE ship_timestamp IS NOT NULL
[ERROR]   TestHivePushdownIntegrationSmokeTest>AbstractTestIntegrationSmokeTest.testAggregateSingleColumn:47->AbstractTestQueryFramework.assertQuery:129->AbstractTestQueryFramework.assertQuery:134 Execution of 'actual' query failed: SELECT MAX(comment) FROM orders
[ERROR]   TestHivePushdownIntegrationSmokeTest>TestHiveIntegrationSmokeTest.testNullBucket:2185->AbstractTestQueryFramework.assertQuery:144 Execution of 'actual' query failed: SELECT COUNT() FROM table_with_null_buckets WHERE key IS NULL
[INFO] 
[ERROR] Tests run: 150, Failures: 11, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

--- a/presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java
@@ -422,6 +422,12 @@ public class HivePageSourceProvider
             }
         }
 
+        if (!hiveColumns.isEmpty() && hiveColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
+            throw new UnsupportedOperationException("Partial aggregation pushdown only supported for ORC/Parquet files. " +
+                    "Table " + tableName.toString() + " has file (" + path.toString() + ") of format " + storage.getStorageFormat().getOutputFormat() +
+                    ". Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again");
+        }
+
         for (HiveRecordCursorProvider provider : cursorProviders) {
             // GenericHiveRecordCursor will automatically do the coercion without HiveCoercionRecordCursor
             boolean doCoercion = !(provider instanceof GenericHiveRecordCursorProvider);

--- a/presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java
@@ -53,7 +53,6 @@ import static com.facebook.presto.hive.HiveSessionProperties.isPartialAggregatio
 import static com.facebook.presto.hive.HiveSessionProperties.isPartialAggregationPushdownForVariableLengthDatatypesEnabled;
 import static com.facebook.presto.hive.HiveStorageFormat.ORC;
 import static com.facebook.presto.hive.HiveStorageFormat.PARQUET;
-import static com.facebook.presto.hive.HiveTableProperties.STORAGE_FORMAT_PROPERTY;
 import static com.facebook.presto.hive.metastore.MetastoreUtil.isArrayType;
 import static com.facebook.presto.hive.metastore.MetastoreUtil.isMapType;
 import static com.facebook.presto.hive.metastore.MetastoreUtil.isRowType;
@@ -125,8 +124,31 @@ public class HivePartialAggregationPushdown
             this.variableAllocator = variableAllocator;
         }
 
-        private boolean isAggregationPushdownSupported(AggregationNode partialAggregationNode, HiveStorageFormat hiveStorageFormat)
+        private boolean isAggregationPushdownSupported(AggregationNode partialAggregationNode)
         {
+            if (partialAggregationNode.hasNonEmptyGroupingSet()) {
+                return false;
+            }
+
+            TableScanNode tableScanNode = (TableScanNode) partialAggregationNode.getSource();
+            ConnectorTableMetadata connectorTableMetadata = metadataFactory.get().getTableMetadata(session, tableScanNode.getTable().getConnectorHandle());
+            Optional<Object> rawFormat = Optional.ofNullable(connectorTableMetadata.getProperties().get(HiveTableProperties.STORAGE_FORMAT_PROPERTY));
+            if (!rawFormat.isPresent()) {
+                return false;
+            }
+
+            final HiveStorageFormat hiveStorageFormat = HiveStorageFormat.valueOf(rawFormat.get().toString());
+            if (hiveStorageFormat != ORC && hiveStorageFormat != PARQUET) {
+                return false;
+            }
+
+            if (tableScanNode.getTable().getLayout().isPresent()) {
+                HiveTableLayoutHandle hiveTableLayoutHandle = (HiveTableLayoutHandle) tableScanNode.getTable().getLayout().get();
+                if (!hiveTableLayoutHandle.getPredicateColumns().isEmpty()) {
+                    return false;
+                }
+            }
+
             /**
              * Aggregation push downs are supported only on primitive types and supported aggregation functions are:
              * count(*), count(columnName), min(columnName), max(columnName)
@@ -159,7 +181,7 @@ public class HivePartialAggregationPushdown
                         return false;
                     }
 
-                    if (hiveStorageFormat == HiveStorageFormat.ORC) {
+                    if (hiveStorageFormat == ORC) {
                         if (TINYINT.equals(type) ||
                                 VARBINARY.equals(type) ||
                                 TIMESTAMP.equals(type)) {
@@ -185,24 +207,12 @@ public class HivePartialAggregationPushdown
             }
 
             AggregationNode partialAggregationNode = (AggregationNode) plan;
-            if (partialAggregationNode.hasNonEmptyGroupingSet()) {
-                return Optional.empty();
-            }
+
             TableScanNode oldTableScanNode = (TableScanNode) partialAggregationNode.getSource();
             TableHandle oldTableHandle = oldTableScanNode.getTable();
             HiveTableHandle hiveTableHandle = getHiveTableHandle(oldTableScanNode).orElseThrow(() -> new PrestoException(NOT_FOUND, "Hive table handle not found"));
 
-            ConnectorTableMetadata connectorTableMetadata = metadataFactory.get().getTableMetadata(session, oldTableHandle.getConnectorHandle());
-            Optional<Object> rawFormat = Optional.ofNullable(connectorTableMetadata.getProperties().get(STORAGE_FORMAT_PROPERTY));
-            if (!rawFormat.isPresent()) {
-                return Optional.empty();
-            }
-            final HiveStorageFormat hiveStorageFormat = HiveStorageFormat.valueOf(rawFormat.get().toString());
-            if (hiveStorageFormat != ORC && hiveStorageFormat != PARQUET) {
-                return Optional.empty();
-            }
-
-            if (!isAggregationPushdownSupported(partialAggregationNode, hiveStorageFormat)) {
+            if (!isAggregationPushdownSupported(partialAggregationNode)) {
                 return Optional.empty();
             }
 

--- a/presto-hive/src/main/java/com/facebook/presto/hive/orc/AggregatedOrcPageSource.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/orc/AggregatedOrcPageSource.java
@@ -138,13 +138,14 @@ public class AggregatedOrcPageSource
             completedBytes += ((FixedWidthType) type).getFixedSize();
         }
 
+        String orcNoMinMaxMessage = "No min/max found for orc file. Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again";
         switch (orcType.getOrcTypeKind()) {
             case SHORT:
             case INT:
             case LONG: {
                 Long value = isMin ? columnStatistics.getIntegerStatistics().getMin() : columnStatistics.getIntegerStatistics().getMax();
                 if (value == null) {
-                    throw new UnsupportedOperationException("No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                    throw new UnsupportedOperationException(orcNoMinMaxMessage);
                 }
                 else {
                     blockBuilder.writeLong(value);
@@ -156,7 +157,7 @@ public class AggregatedOrcPageSource
             case DATE: {
                 Integer value = isMin ? columnStatistics.getDateStatistics().getMin() : columnStatistics.getDateStatistics().getMax();
                 if (value == null) {
-                    throw new UnsupportedOperationException("No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                    throw new UnsupportedOperationException(orcNoMinMaxMessage);
                 }
                 else {
                     blockBuilder.writeLong(Long.valueOf(value));
@@ -169,7 +170,7 @@ public class AggregatedOrcPageSource
             case STRING: {
                 Slice value = isMin ? columnStatistics.getStringStatistics().getMin() : columnStatistics.getStringStatistics().getMax();
                 if (value == null) {
-                    throw new UnsupportedOperationException("No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                    throw new UnsupportedOperationException(orcNoMinMaxMessage);
                 }
                 else {
                     blockBuilder.writeBytes(value, 0, value.length()).closeEntry();
@@ -181,7 +182,7 @@ public class AggregatedOrcPageSource
             case FLOAT: {
                 Double value = isMin ? columnStatistics.getDoubleStatistics().getMin() : columnStatistics.getDoubleStatistics().getMax();
                 if (value == null) {
-                    throw new UnsupportedOperationException("No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                    throw new UnsupportedOperationException(orcNoMinMaxMessage);
                 }
                 else {
                     blockBuilder.writeLong(floatToRawIntBits(value.floatValue()));
@@ -192,7 +193,7 @@ public class AggregatedOrcPageSource
             case DOUBLE: {
                 Double value = isMin ? columnStatistics.getDoubleStatistics().getMin() : columnStatistics.getDoubleStatistics().getMax();
                 if (value == null) {
-                    throw new UnsupportedOperationException("No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                    throw new UnsupportedOperationException(orcNoMinMaxMessage);
                 }
                 else {
                     type.writeDouble(blockBuilder, value);
@@ -203,7 +204,7 @@ public class AggregatedOrcPageSource
             case DECIMAL:
                 BigDecimal value = isMin ? columnStatistics.getDecimalStatistics().getMin() : columnStatistics.getDecimalStatistics().getMax();
                 if (value == null) {
-                    throw new UnsupportedOperationException("No min/max found for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                    throw new UnsupportedOperationException(orcNoMinMaxMessage);
                 }
                 else {
                     Type definedType = hiveType.getType(typeManager);
@@ -232,7 +233,7 @@ public class AggregatedOrcPageSource
     {
         ColumnStatistics columnStatistics = footer.getFileStats().get(columnIndex + 1);
         if (!columnStatistics.hasNumberOfValues()) {
-            throw new UnsupportedOperationException("Number of values not set for orc file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+            throw new UnsupportedOperationException("Number of values not set for orc file. Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again");
         }
         blockBuilder.writeLong(columnStatistics.getNumberOfValues());
     }

--- a/presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSourceFactory.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSourceFactory.java
@@ -270,7 +270,7 @@ public class OrcBatchPageSourceFactory
                 }
             }
 
-            if (!physicalColumns.isEmpty() && physicalColumns.get(0).getColumnType() == AGGREGATED) {
+            if (!physicalColumns.isEmpty() && physicalColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
                 return new AggregatedOrcPageSource(physicalColumns, reader.getFooter(), typeManager, functionResolution);
             }
 

--- a/presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java
@@ -335,7 +335,7 @@ public class OrcSelectivePageSourceFactory
 
             List<HiveColumnHandle> physicalColumns = getPhysicalHiveColumnHandles(columns, useOrcColumnNames, reader, path);
 
-            if (!physicalColumns.isEmpty() && physicalColumns.get(0).getColumnType() == AGGREGATED) {
+            if (!physicalColumns.isEmpty() && physicalColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
                 return new AggregatedOrcPageSource(physicalColumns, reader.getFooter(), typeManager, functionResolution);
             }
 

--- a/presto-hive/src/main/java/com/facebook/presto/hive/parquet/AggregatedParquetPageSource.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/parquet/AggregatedParquetPageSource.java
@@ -107,21 +107,21 @@ public class AggregatedParquetPageSource
             Aggregation aggregation = columnHandle.getPartialAggregation().get();
             Type type = typeManager.getType(columnHandle.getTypeSignature());
             BlockBuilder blockBuilder = type.createBlockBuilder(null, batchSize, 0);
-            String inputColumn = aggregation.getArguments().isEmpty() ? null : aggregation.getArguments().get(0).toString();
+            int columnIndex = columnHandle.getHiveColumnIndex();
             FunctionHandle functionHandle = aggregation.getFunctionHandle();
 
             if (functionResolution.isCountFunction(functionHandle)) {
                 long rowCount = getRowCountFromParquetMetadata(parquetMetadata);
-                if (inputColumn != null) {
-                    rowCount -= getNumNulls(parquetMetadata, inputColumn);
+                if (!aggregation.getArguments().isEmpty()) {
+                    rowCount -= getNumNulls(parquetMetadata, columnIndex);
                 }
                 blockBuilder = blockBuilder.writeLong(rowCount);
             }
             else if (functionResolution.isMaxFunction(functionHandle)) {
-                writeMinMax(parquetMetadata, inputColumn, blockBuilder, type, columnHandle.getHiveType(), false);
+                writeMinMax(parquetMetadata, columnIndex, blockBuilder, type, columnHandle.getHiveType(), false);
             }
             else if (functionResolution.isMinFunction(functionHandle)) {
-                writeMinMax(parquetMetadata, inputColumn, blockBuilder, type, columnHandle.getHiveType(), true);
+                writeMinMax(parquetMetadata, columnIndex, blockBuilder, type, columnHandle.getHiveType(), true);
             }
             else {
                 throw new UnsupportedOperationException(aggregation.getFunctionHandle().toString() + " is not supported");
@@ -144,14 +144,13 @@ public class AggregatedParquetPageSource
         return rowCount;
     }
 
-    private long getNumNulls(ParquetMetadata parquetMetadata, String columnName)
+    private long getNumNulls(ParquetMetadata parquetMetadata, int columnIndex)
     {
         long numNulls = 0;
-        int columnIndex = parquetMetadata.getFileMetaData().getSchema().getFieldIndex(columnName);
         for (BlockMetaData blockMetaData : parquetMetadata.getBlocks()) {
             Statistics statistics = blockMetaData.getColumns().get(columnIndex).getStatistics();
             if (!statistics.isNumNullsSet()) {
-                throw new UnsupportedOperationException("Number of nulls not set for parquet file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                throw new UnsupportedOperationException("Number of nulls not set for parquet file. Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again");
             }
             numNulls += statistics.getNumNulls();
         }
@@ -159,9 +158,8 @@ public class AggregatedParquetPageSource
         return numNulls;
     }
 
-    private void writeMinMax(ParquetMetadata parquetMetadata, String columnName, BlockBuilder blockBuilder, Type type, HiveType hiveType, boolean isMin)
+    private void writeMinMax(ParquetMetadata parquetMetadata, int columnIndex, BlockBuilder blockBuilder, Type type, HiveType hiveType, boolean isMin)
     {
-        int columnIndex = parquetMetadata.getFileMetaData().getSchema().getFieldIndex(columnName);
         org.apache.parquet.schema.Type parquetType = parquetMetadata.getFileMetaData().getSchema().getType(columnIndex);
         if (parquetType instanceof GroupType) {
             throw new IllegalArgumentException("Unsupported type : " + parquetType.toString());
@@ -171,7 +169,7 @@ public class AggregatedParquetPageSource
         for (BlockMetaData blockMetaData : parquetMetadata.getBlocks()) {
             Statistics statistics = blockMetaData.getColumns().get(columnIndex).getStatistics();
             if (!statistics.hasNonNullValue()) {
-                throw new UnsupportedOperationException("No min/max found for parquet file. Set session property pushdown_partial_aggregations_into_scan=false and execute query again");
+                throw new UnsupportedOperationException("No min/max found for parquet file. Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again");
             }
             if (isMin) {
                 Object currentValue = statistics.genericGetMin();

--- a/presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java
@@ -206,7 +206,7 @@ public class ParquetPageSourceFactory
         try {
             FSDataInputStream inputStream = hdfsEnvironment.getFileSystem(user, path, configuration).openFile(path, hiveFileContext);
             ParquetMetadata parquetMetadata = MetadataReader.readFooter(inputStream, path, fileSize);
-            if (!columns.isEmpty() && columns.get(0).getColumnType() == AGGREGATED) {
+            if (!columns.isEmpty() && columns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
                 return new AggregatedParquetPageSource(columns, parquetMetadata, typeManager, functionResolution);
             }
 

--- a/presto-hive/src/main/java/com/facebook/presto/hive/rcfile/RcFilePageSourceFactory.java
+++ b/presto-hive/src/main/java/com/facebook/presto/hive/rcfile/RcFilePageSourceFactory.java
@@ -56,6 +56,7 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.Properties;
 
+import static com.facebook.presto.hive.HiveColumnHandle.ColumnType.AGGREGATED;
 import static com.facebook.presto.hive.HiveErrorCode.HIVE_BAD_DATA;
 import static com.facebook.presto.hive.HiveErrorCode.HIVE_CANNOT_OPEN_SPLIT;
 import static com.facebook.presto.hive.HiveErrorCode.HIVE_MISSING_DATA;
@@ -110,6 +111,12 @@ public class RcFilePageSourceFactory
             HiveFileContext hiveFileContext,
             Optional<EncryptionInformation> encryptionInformation)
     {
+        if (!columns.isEmpty() && columns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
+            throw new UnsupportedOperationException("Partial aggregation pushdown only supported for ORC/Parquet files. " +
+                    "Table " + tableName.toString() + " has file (" + path.toString() + ") of format " + storage.getStorageFormat().getOutputFormat() +
+                    ". Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again");
+        }
+
         RcFileEncoding rcFileEncoding;
         if (LazyBinaryColumnarSerDe.class.getName().equals(storage.getStorageFormat().getSerDe())) {
             rcFileEncoding = new BinaryRcFileEncoding();

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownDistributedQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownDistributedQueries.java
@@ -32,11 +32,15 @@ public class TestHivePushdownDistributedQueries
     public TestHivePushdownDistributedQueries()
     {
         super(() -> createQueryRunner(
-                    getTables(),
-                    ImmutableMap.of("experimental.pushdown-subfields-enabled", "true", "experimental.pushdown-dereference-enabled", "true"),
-                    "sql-standard",
-                    ImmutableMap.of("hive.pushdown-filter-enabled", "true", "hive.enable-parquet-dereference-pushdown", "true"),
-                    Optional.empty()));
+                getTables(),
+                ImmutableMap.of("experimental.pushdown-subfields-enabled", "true",
+                        "experimental.pushdown-dereference-enabled", "true"),
+                "sql-standard",
+                ImmutableMap.of("hive.pushdown-filter-enabled", "true",
+                        "hive.enable-parquet-dereference-pushdown", "true",
+                        "hive.partial_aggregation_pushdown_enabled", "true",
+                        "hive.partial_aggregation_pushdown_for_variable_length_datatypes_enabled", "true"),
+                Optional.empty()));
     }
 
     @Override

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java
@@ -42,6 +42,7 @@ import static com.facebook.presto.common.type.StandardTypes.SMALLINT;
 import static com.facebook.presto.common.type.StandardTypes.TINYINT;
 import static com.facebook.presto.common.type.StandardTypes.VARCHAR;
 import static com.facebook.presto.hive.HiveQueryRunner.HIVE_CATALOG;
+import static com.facebook.presto.hive.HiveSessionProperties.PARTIAL_AGGREGATION_PUSHDOWN_ENABLED;
 import static com.facebook.presto.hive.HiveSessionProperties.PUSHDOWN_FILTER_ENABLED;
 import static com.facebook.presto.hive.HiveStorageFormat.RCBINARY;
 import static com.facebook.presto.hive.HiveStorageFormat.RCTEXT;
@@ -103,9 +104,13 @@ public class TestHivePushdownFilterQueries
             throws Exception
     {
         DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(getTables(),
-                ImmutableMap.of("experimental.pushdown-subfields-enabled", "true", "experimental.pushdown-dereference-enabled", "true"),
+                ImmutableMap.of("experimental.pushdown-subfields-enabled", "true",
+                        "experimental.pushdown-dereference-enabled", "true"),
                 "sql-standard",
-                ImmutableMap.of("hive.pushdown-filter-enabled", "true", "hive.enable-parquet-dereference-pushdown", "true"),
+                ImmutableMap.of("hive.pushdown-filter-enabled", "true",
+                        "hive.enable-parquet-dereference-pushdown", "true",
+                        "hive.partial_aggregation_pushdown_enabled", "true",
+                        "hive.partial_aggregation_pushdown_for_variable_length_datatypes_enabled", "true"),
                 Optional.empty());
 
         queryRunner.execute(noPushdownFilter(queryRunner.getDefaultSession()),
@@ -1034,7 +1039,8 @@ public class TestHivePushdownFilterQueries
             // no filter
             assertQueryUsingH2Cte("SELECT * FROM test_file_format_orc", cte);
             assertQueryUsingH2Cte("SELECT comment FROM test_file_format_orc", cte);
-            assertQueryUsingH2Cte("SELECT COUNT(*) FROM test_file_format_orc", cte);
+            assertQueryFails("SELECT COUNT(*) FROM test_file_format_orc", "Partial aggregation pushdown only supported for ORC/Parquet files. Table tpch.test_file_format_orc has file ((.*?)) of format (.*?). Set session property hive.pushdown_partial_aggregations_into_scan=false and execute query again");
+            assertQueryUsingH2Cte(noPartialAggregationPushdown(queryRunner.getDefaultSession()), "SELECT COUNT(*) FROM test_file_format_orc", cte, Function.identity());
 
             // filter on partition column
             assertQueryUsingH2Cte("SELECT comment from test_file_format_orc WHERE ds='2019-11-01'", cte);
@@ -1134,6 +1140,11 @@ public class TestHivePushdownFilterQueries
         assertQueryUsingH2Cte(query, Function.identity());
     }
 
+    private void assertQueryUsingH2Cte(Session session, String query, String cte, Function<String, String> rewriter)
+    {
+        assertQuery(session, query, cte + toH2(rewriter.apply(query)));
+    }
+
     private void assertQueryUsingH2Cte(String query, Function<String, String> rewriter)
     {
         assertQuery(query, WITH_LINEITEM_EX + toH2(rewriter.apply(query)));
@@ -1168,6 +1179,13 @@ public class TestHivePushdownFilterQueries
     {
         return Session.builder(session)
                 .setCatalogSessionProperty(HIVE_CATALOG, PUSHDOWN_FILTER_ENABLED, "false")
+                .build();
+    }
+
+    private static Session noPartialAggregationPushdown(Session session)
+    {
+        return Session.builder(session)
+                .setCatalogSessionProperty(HIVE_CATALOG, PARTIAL_AGGREGATION_PUSHDOWN_ENABLED, "false")
                 .build();
     }
 }

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownIntegrationSmokeTest.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownIntegrationSmokeTest.java
@@ -36,9 +36,14 @@ public class TestHivePushdownIntegrationSmokeTest
     {
         super(() -> createQueryRunner(
                     ImmutableList.of(ORDERS, CUSTOMER, LINE_ITEM, PART_SUPPLIER),
-                    ImmutableMap.of("experimental.pushdown-subfields-enabled", "true", "experimental.pushdown-dereference-enabled", "true"),
+                    ImmutableMap.of("experimental.pushdown-subfields-enabled", "true",
+                            "experimental.pushdown-dereference-enabled", "true"),
                     "sql-standard",
-                    ImmutableMap.of("hive.pushdown-filter-enabled", "true", "hive.enable-parquet-dereference-pushdown", "true"),
+                    ImmutableMap.of("hive.pushdown-filter-enabled", "true",
+                            "hive.enable-parquet-dereference-pushdown", "true",
+                            "hive.partial_aggregation_pushdown_enabled", "true",
+                            "hive.partial_aggregation_pushdown_for_variable_length_datatypes_enabled", "true",
+                            "hive.orc.writer.string-statistics-limit", "128B"),
                     Optional.empty()),
                 createBucketedSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))),
                 createMaterializeExchangesSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))),

--- a/presto-hive/src/test/java/com/facebook/presto/hive/TestParquetDistributedQueries.java
+++ b/presto-hive/src/test/java/com/facebook/presto/hive/TestParquetDistributedQueries.java
@@ -43,6 +43,8 @@ public class TestParquetDistributedQueries
                 .put("hive.parquet.use-column-names", "true")
                 .put("hive.compression-codec", "GZIP")
                 .put("hive.enable-parquet-dereference-pushdown", "true")
+                .put("hive.partial_aggregation_pushdown_enabled", "true")
+                .put("hive.partial_aggregation_pushdown_for_variable_length_datatypes_enabled", "true")
                 .build();
         return HiveQueryRunner.createQueryRunner(
                 getTables(),

